mtp_api/apps/performance/tests/test_views.py

```diff
@@ -0,0 +1,122 @@
+import datetime
+
+from django.urls import reverse
+from django.utils import timezone
+from model_mommy import mommy
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from core.tests.utils import make_test_users
+from core.utils import monday_of_same_week
+from mtp_auth.tests.utils import AuthTestCaseMixin
+from performance.models import PerformanceData
+
+
+class PerformanceDataViewTestCase(AuthTestCaseMixin, APITestCase):
+    fixtures = ['initial_types.json', 'test_prisons.json', 'initial_groups.json']
+
+    def setUp(self):
+        super().setUp()
+
+        test_users = make_test_users()
+        self.send_money_user = test_users['send_money_users'][0]
+        self.bank_admin_user = test_users['bank_admins'][0]
+        self.prison_clerk_user = test_users['prison_clerks'][0]
+
+    @property
+    def performance_data_url(self):
+        return reverse('performance-data')
+
+    def test_only_accessible_from_send_money(self):
+        for (user, expected_status_code) in [
+            (self.send_money_user, status.HTTP_200_OK),
+            (self.bank_admin_user, status.HTTP_403_FORBIDDEN),
+            (self.prison_clerk_user, status.HTTP_403_FORBIDDEN),
+        ]:
+            http_auth_header = self.get_http_authorization_for_user(user)
+            response = self.client.get(
+                self.performance_data_url, data={}, format='json',
+                HTTP_AUTHORIZATION=http_auth_header
+            )
+            self.assertEqual(response.status_code, expected_status_code,
+                             f'expected {expected_status_code} for user {user}')
+
+    def test_headers(self):
+        response = self.client.get(
+            self.performance_data_url, data={}, format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(self.send_money_user),
+        )
+
+        self.assertDictEqual(
+            response.json()['headers'],
+            {
+                'week': 'Week commencing',
+                'credits_total': 'Transactions – total',
+                'credits_by_mtp': 'Transactions – online',
+                'digital_takeup': 'Digital take-up',
+                'completion_rate': 'Completion rate',
+                'user_satisfaction': 'User satisfaction',
+                'rated_1': 'Very dissatisfied',
+                'rated_2': 'Dissatisfied',
+                'rated_3': 'Neither satisfied or dissatisfied',
+                'rated_4': 'Satisfied',
+                'rated_5': 'Very satisfied',
+            }
+        )
+
+    def test_percentages_are_formatted(self):
+        mommy.make(
+            PerformanceData,
+            week=self._monday_n_weeks_ago(4),
+            digital_takeup=0.95,
+            completion_rate=0.8111,
+            user_satisfaction=0.96666666,
```

Comment:
i would also try with another model with a `0` and a `None` somewhere. i'd expect `'0%'` and `''` to come out of it

Reply:
https://github.com/ministryofjustice/money-to-prisoners-api/pull/669/commits/d1c291f9817956393506d45309bdb495c2bf0228